client/src/components/Upload/utils.js

```diff
@@ -1,7 +1,128 @@
 /*
  * Utilities for working with upload data structures.
  */
+import { errorMessageAsString, rethrowSimple } from "utils/simple-error";
 
+import { getDatatypes, getGenomes, getRemoteFiles } from "./services";
+
+export const AUTO_EXTENSION = {
+    id: "auto",
+    text: "Auto-detect",
+    description:
+        "This system will try to detect the file type automatically. If your file is not detected properly as one of the known formats, it most likely means that it has some format problems (e.g., different number of columns on different rows). You can still coerce the system to set your data to the format you think it should be.  You can also upload compressed files, which will automatically be decompressed.",
+};
+export const DEFAULT_DBKEY = "?";
+export const DEFAULT_EXTENSION = "auto";
+
+/**
+ * Local cache.
+ */
+let _cachedDatatypes;
+let _cachedDbKeys;
```

Comment:
Instead of adding our own cache system it may be a better option to use the store right away ?

Reply:
Yes, I'll make a note to move this too a store in a follow-ups.